web/server/codechecker_server/migrations/report/versions/75ae226b5d88_review_status_for_each_report.py

```diff
@@ -0,0 +1,157 @@
+            SET (review_status, review_status_author, review_status_date, review_status_message) =
+                (SELECT status, author, date, message FROM review_statuses WHERE bug_hash = reports.bug_id)
+        """)
+        conn.execute("""
+            UPDATE reports
+            SET review_status = 'unreviewed'
+            WHERE review_status IS NULL
+        """)
+    elif dialect == 'postgresql':
+        op.add_column('reports', col_rs)
+        op.add_column('reports', col_rs_author)
+        op.add_column('reports', col_rs_date)
+        op.add_column('reports', col_rs_is_in_source)
+        op.add_column('reports', col_rs_message)
+
+        conn.execute("""
+            UPDATE reports
+            SET (review_status, review_status_author, review_status_date, review_status_message) =
+                (SELECT status, author, date, message FROM review_statuses WHERE bug_hash = reports.bug_id)
+        """)
+        conn.execute("""
+            UPDATE reports
+            SET review_status = 'unreviewed'
+            WHERE review_status IS NULL
+        """)
+
+    source_code_comment_handler = SourceCodeCommentHandler()
+
+    files_with_report = conn.execute("""
+        SELECT reports.file_id, files.content_hash
+        FROM reports INNER JOIN files ON reports.file_id = files.id
+    """)
+
+    review_status_in_source = set()
+
+    for files in util.chunks(files_with_report, FILE_CHUNK_SIZE):
+        content_hashes = set()
+        file_ids = set()
+        file_id_to_content_hash = {}
+
+        for f in files:
+            content_hashes.add(f"'{f.content_hash}'")
+            file_ids.add(f"'{f.file_id}'")
+            file_id_to_content_hash[f.file_id] = f.content_hash
+
+        content_hashes = ','.join(content_hashes)
+        file_ids = ','.join(file_ids)
+
+        hash_to_content = conn.execute(f"""
+            SELECT content_hash, content FROM file_contents
+            WHERE content_hash IN ({content_hashes})
+        """)
+
+        hash_to_content = {
+            x.content_hash: decode_file_content(x.content)
+            for x in hash_to_content}
+
+        report_id_to_line = conn.execute(f"""
+            SELECT id, file_id, checker_id, line FROM reports
+            WHERE file_id IN ({file_ids})
+        """)
+
+        for row in report_id_to_line:
+            comments = source_code_comment_handler.get_source_line_comments(
+                StringIO(
+                    hash_to_content[file_id_to_content_hash[row.file_id]]),
+                row.line)
+
+            for comment in comments:
+                if row.checker_id in comment.checkers or \
+                        'all' in comment.checkers:
+                    review_status_in_source.add(row.id)
+
+    if review_status_in_source:
+        conn.execute(f"""
+            UPDATE reports
+            SET review_status_is_in_source = '1'
+            WHERE id IN ({','.join(map(str, review_status_in_source))})
+        """)
+
```

Comment:
Create a `DELETE SQL` query to remove entries from the global review status table where there is at least one report which has a source code comment.